src/SimpleSocket.cpp

```diff
@@ -425,6 +426,9 @@ bool CSimpleSocket::Open(const char *pAddr, uint16 nPort)
     }
     case CSimpleSocket::SocketTypeRaw :
         break;
+    case CSimpleSocket::SocketTypeInvalid:
+    case CSimpleSocket::SocketTypeTcp6:
+    case CSimpleSocket::SocketTypeUdp6:
     default:
```

Comment:
Same here for the Tcp6 and Udp6 cases (and below). Not sure if they should be handled the same way as Tcp/Udp or not, though (I'm assuming you would know better).